lib/db.c

```diff
@@ -128,6 +129,20 @@ gen_pathname(const char *path, char *buffer, int fno)
   }
 }
 
+static uint32_t target_last_modified = 0;
```

Comment:
This scope of a variable is too large.
I think a static variable is not threaded safe.

We should scope of a variable is local in function.

Reply:
fixed.